src/extension/content.js

```diff
@@ -42,13 +42,23 @@
         'devOrigin',
         'adminVersion',
         'authTokenExpiry',
+        'transient',
       ].includes(k)));
     curatedConfig.scriptUrl = chrome.runtime.getURL('index.js');
 
     sidekick = new AEMSidekick(curatedConfig);
     sidekick.setAttribute('open', `${display}`);
     document.body.prepend(sidekick);
     window.hlx.sidekick = sidekick;
+
+    sidekick.addEventListener('projectadded', () => {
+      chrome.runtime.sendMessage({ action: 'addRemoveProject', url: window.location.href });
+    });
+
+    sidekick.addEventListener('hidden', () => {
+      toggleDisplay();
+      chrome.runtime.sendMessage({ action: 'updateUI' });
```

Comment:
I'd not send this message...

Reply:
We need to both toggle the display and update the extension icon in the browser (using the background script). I'll look at your suggestion to use chrome.storage.local.onChange though. I like that better.